README.md

```diff
@@ -0,0 +1,120 @@
+# Amazon EC2 Instance Identity Document Verification
+This is a small node module written in C using N-API and the OpenSSL C API to
+perform instance identity document verification.
+
+# Usage
+This library has a very simple interface:
+
+```javascript
+let fs = require('fs');
+let verify = require('iid-verify');
+
+// http://169.254.169.254/latest/dynamic/instance-identity/{rsa2048, document}
+// Public key provided by Amazon
+
+// document is a JSON file available through the metadata service which
+// contains information about the instance like instance id and AMI used to
+// configure the instance
+let document = fs.readFileSync('document');
+
+// rsa2048 is a PCKS#7 envelope which contains the cryptographic signature
+// of the document computed against the Amazon EC2 private key.  We strongly
+// suggest using the rsa2048 enpoint instead of the pkcs7 endpoint because
+// the rsa2048 endpoint uses SHA256 instead of the pkcs7 endpoint's SHA1
+let rsa2048 = fs.readFileSync('rsa2048');
+
+// pubkey is a PEM encoded public key as provided by Amazon EC2.  This
+// public key is unique to each region and each account and so must be
+// obtained by requesting in a service ticket
+let pubkey = fs.readFileSync('pubkey');
+
+if (verify(pubkey, document, rsa2048)){
+  console.log('This document is valid!');
+} else {
+  console.log('CAUTION: This document is INVALID!');
+}
+```
+
+# API
+There is a single function in this library: `verify`.  It takes a public key as
+provided by Amazon, a clear text copy of the instance identity document and the
+`rsa2048` metadata endpoint value.  The `rsa2048` document from the metadata
+service is a PKCS#7 envelope without the PEM headers (e.g. `------XXX------`).
+
+This library can handle with and without the PEM headers for the `pkcs7`
+argument.
+
+# Errors
+The `verify` function of this library has three expected outcomes:
+
+1. Document validates against signature and public key -- instance identity
+   document is valid and trustworthy.  `verify` returns true.
+2. Document does not validate against signature -- instance identity document,
+   public key or pkcs7 signature is not correct.  `verify` returns false.
+3. A javascript `Error` is thrown from the library along with information about
+   which error occurred.
```

Comment:
nit: seems complicated to return `isValid` and throw errors. Is returning an array of errors or throwing an "invalid signature" error an option?

Reply:
It's been a taskcluster pattern to have checks like these use a boolean true/false to represent the two successful validation cases and thrown errors only when the validation itself could not be completed.  I tend to agree with you, but I wanted to match conventions elsewhere in our code.